### ISSUE TITLE
test-filters error writes empty database named INVALID! to Duplicati current directory

### ISSUE THREAD

**Issue by ts678** [duplicati/duplicati#5091]
- [x] I have searched open and closed issues for duplicates.
- [x] I have searched the [forum](https://forum.duplicati.com) for related topics.

----------------------------------------

## Environment info

- **Duplicati version**: 2.0.7.100 but seemingly not before
- **Operating system**: Windows 10
- **Backend**: N/A

## Description

Near my install folder was an SQLite database where (in SQL dump) the only data content was `INSERT INTO "Version" VALUES (1,12);`
I looked to see what might set a database name like that, and think it might be from TestFilter(), but I haven't been able to force the bug.

https://github.com/duplicati/duplicati/blob/e7c191389cdb3fdb9500d83ebd06e0ae99634fd1/Duplicati/Library/Main/Controller.cs#L305-L308

The `Everything` search utility for Windows gives essentially instant results, so I found another in a test build, and found its profiling log:

```
2023-12-17 10:36:37 -05 - [Information-Duplicati.Library.Main.Controller-StartingOperation]: The operation TestFilters has started
2023-12-17 10:36:37 -05 - [Profiling-Timer.Begin-Duplicati.Library.Main.Controller-RunTestFilters]: Starting - Running TestFilters
2023-12-17 10:36:38 -05 - [Profiling-Timer.Finished-Duplicati.Library.Main.Controller-RunTestFilters]: Running TestFilters took 0:00:00:00.190
2023-12-17 10:36:44 -05 - [Profiling-Timer.Begin-Duplicati.Library.Main.Database.ExtensionMethods-ExecuteReader]: Starting - ExecuteReader: SELECT "ID", "Timestamp" FROM "Operation" ORDER BY "Timestamp" DESC LIMIT 1
2023-12-17 10:36:44 -05 - [Profiling-Timer.Finished-Duplicati.Library.Main.Database.ExtensionMethods-ExecuteReader]: ExecuteReader: SELECT "ID", "Timestamp" FROM "Operation" ORDER BY "Timestamp" DESC LIMIT 1 took 0:00:00:00.000
2023-12-17 10:36:44 -05 - [Error-Duplicati.Library.Main.Controller-FailedOperation]: The operation TestFilters has failed with error: Invalid path: "file://C:\Duplicati\duplicati-2.0.7.3_canary_2023-12-13.gpatel-fr\RUN\test 1\" (The given path's format is not supported.)
Duplicati.Library.Interface.UserInformationException: Invalid path: "file://C:\Duplicati\duplicati-2.0.7.3_canary_2023-12-13.gpatel-fr\RUN\test 1\" (The given path's format is not supported.) ---> System.NotSupportedException: The given path's format is not supported.
   at System.Security.Permissions.FileIOPermission.EmulateFileIOPermissionChecks(String fullPath)
   at System.Security.Permissions.FileIOPermission.QuickDemand(FileIOPermissionAccess access, String fullPath, Boolean checkForDuplicates, Boolean needFullPath)
   at Duplicati.Library.Main.Controller.ExpandInputSources(String[] inputsources, IFilter filter)
   --- End of inner exception stack trace ---
   at Duplicati.Library.Main.Controller.ExpandInputSources(String[] inputsources, IFilter filter)
   at Duplicati.Library.Main.Controller.<>c__DisplayClass31_0.<TestFilter>b__1(TestFilterResults result)
   at Duplicati.Library.Main.Controller.RunAction[T](T result, String[]& paths, IFilter& filter, Action`1 method)
```

Explorer says the DB creation time is December ‎17, ‎2023, ‏‎10:36:38 AM and the modification time is ‎December ‎17, ‎2023, ‏‎10:36:44 AM.
Location of this first one is in the RUN folder above. The 2.0.7.100 is in its top level install folder. Possibly both were a current directory.

That could invalidate an update by throwing off its verification. I don't find any older such odd files, so I wonder if this is a regression...
There were several pull requests involving filters that got into 2.0.7.100. I don't know what would have been in the gpatel-fr test build.

2.0.7.100 TrayIcon also runs a profiling log, but I couldn't find anything interesting going on at time of the INVALID! database creation. Possibly I was usng Duplicati by hand at the time, which would also explain why the RUN.bat script for TrayIcon didn't change directory.

## Steps to reproduce

1. Unknown. From code (though I can't find all) and log, it looks like it must `test-filters`, but CLI or GUI Commandline won't do it.

- **Actual result**:
Folder polluted with INVALID! file.
- **Expected result**:
Folder left alone.

**Comment by Jojo-1000** [comment]
This might be a result of the new error reporting. The test command fails due to an error (not sure what caused yours, but shouldn't matter), but no DB to log it to. So when it tries, it will create that database.

Maybe try a test-filter in a folder that has no read permissions to reproduce.

Maybe we should check if the database exists first before logging the error.

**Comment by ts678** [comment]
> Maybe try a test-filter in a folder that has no read permissions to reproduce.

That didn't do it, but invalid drive letter did. I also had Sysinternals Process Monitor watching for activity on that file name.
After second test below, I did another (not shown) and saw that for CLI it goes to my Command Prompt current directory.

```
C:\Duplicati\duplicati-2.0.7.100_canary_2023-12-27>Duplicati.CommandLine.exe test-filters C:\Windows\System32\config\
Including source path: C:\Windows\System32\config\
Error reported while accessing file: C:\Windows\System32\config\ => Access to the path '\\?\C:\Windows\System32\config' is denied.
Error reported while accessing file: C:\Windows\System32\config\ => Access to the path '\\?\C:\Windows\System32\config' is denied.
Including source path: C:\Windows\System32\config\
Error reported while accessing file: C:\Windows\System32\config\ => Access to the path '\\?\C:\Windows\System32\config' is denied.
Error reported while accessing file: C:\Windows\System32\config\ => Access to the path '\\?\C:\Windows\System32\config' is denied.
Including source path: C:\Windows\System32\config\
Matched 0 files (0 bytes)

C:\Duplicati\duplicati-2.0.7.100_canary_2023-12-27>Duplicati.CommandLine.exe test-filters Y:\Windows\System32\config\

System.IO.IOException: Backup aborted since the source path Y:\Windows\System32\config\ does not exist.  Please verify that the source path exists, or remove the source path from the backup configuration, or set the allow-missing-source option.
   at Duplicati.Library.Main.Controller.ExpandInputSources(String[] inputsources, IFilter filter)
   at Duplicati.Library.Main.Controller.<>c__DisplayClass31_0.<TestFilter>b__1(TestFilterResults result)
   at Duplicati.Library.Main.Controller.RunAction[T](T result, String[]& paths, IFilter& filter, Action`1 method)
   at Duplicati.Library.Main.Controller.TestFilter(String[] paths, IFilter filter)
   at Duplicati.CommandLine.Commands.TestFilters(TextWriter outwriter, Action`1 setup, List`1 args, Dictionary`2 options, IFilter filter)
   at Duplicati.CommandLine.Program.ParseCommandLine(TextWriter outwriter, Action`1 setup, Boolean& verboseErrors, String[] args)
   at Duplicati.CommandLine.Program.RunCommandLine(TextWriter outwriter, TextWriter errwriter, Action`1 setup, String[] args)

C:\Duplicati\duplicati-2.0.7.100_canary_2023-12-27>
```

**Comment by ts678** [comment]
The accident can be blocked by permissions. If I do a desktop icon start of 2.0.6.3 with a 2.0.7.100 upgrade, current directory is C:\Program Files\Duplicati 2\ so in this case it can't leave the errant file, even though Process Monitor shows that it's trying to...

![image](https://github.com/duplicati/duplicati/assets/40771998/202f690c-b914-4dc3-8062-7da48714842b)

**Comment by ts678** [comment]
The dbpath path set to `INVALID!` is really old, and seems unique to `test-filters`, which is why I updated the Issue title.

More generally speaking though, there are several CLI commands that don't intentionally involve a database, and there's a situation early on in the life of a server job where the database is assigned but there's not actually a database put there yet.

Commands that don't intentionally involve a database might get a `dbpath` anyway, if set up using the GUI Commandline.

I notice that there are some old issues talking about LogData, but I'm now also wondering how Operation table is handled.

Those are my thoughts on the general database issue. The goal for the new logging is probably to make no situation worse.

**Comment by ts678** [comment]
Despite good testing news from the desktop shortcut, this file breaks a service install running an update, by failing verification.

![image](https://github.com/duplicati/duplicati/assets/40771998/ff024a73-2478-4314-9e0a-df1c1a5a10c1)

[Windows service install/uninstall/start invalidates update by writing into it, causing version downgrade #5088](https://github.com/duplicati/duplicati/issues/5088) explains how so.
In the case here, the 2.0.7.100 that was running before could not come up on service restart. It downgraded to the next best...
